tools/as/as.cpp

```diff
@@ -121,7 +121,8 @@ int main(int argc, char** argv) {
     return error;
   }
 
-  if (FILE* fp = fopen(outFile, "wb")) {
+  const bool use_stdout = *outFile == '-' && *(outFile + 1) == 0;
```

Comment:
Alternate could have written outFile[0] and outFile[1].
But this is ok too.  Doesn't slow me down.